client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -721,19 +730,9 @@ export class CheckoutThankYou extends Component {
 						/>
 					) }
 
-					{ ! isSimplified && ComponentClass && (
+					{ ! isSimplified && ! this.isGenericReceipt() && (
```

Comment:
This refactor may be wrong. I can't recall how I came up with it but this prevents the generic domain registration component from being rendered, amongst other things, I suspect.

Reply:
Upon review, I actually think this logic is correct. However, to make the refactor easier to read, I've restored the original behavior by using a string in place of `ComponentClass`.